core/utils.h

```diff
@@ -1946,6 +1946,14 @@ notify(syslog_event_type_t priority, bool internal, bool synch,
 #define SYSLOG_INTERNAL_CRITICAL_ONCE(...) \
     DODEBUG_ONCE(SYSLOG_INTERNAL_CRITICAL(__VA_ARGS__))
 
+#define MSG_COULDNT_SYNCH_WITH_ALL_THREADS_CODE 42
```

Comment:
If we're going to have separate integer exit codes, let's generate them in CMake_events.cmake.

If these codes are not all that useful, I would say do not add the fatal error macro, and just do a SYSLOG and os_terminate at the site, and ensure os_terminate's default code is non-zero

Reply:
I think a unique message will work well for now, let's just go with that rather than invent something new.